### PR TITLE
Global Total Integrated primary production derived variable

### PR DESCRIPTION
**Pull request by ledm**
This PR brings in the Global Total Integrated primary production derived variable. This variable has appeared in the upcoming ESMValTool Paper - part 2, so it would be nice to get it out there before the publication gets to discussions format. 

This is linked with the ESMValTool PR https://github.com/ESMValGroup/ESMValTool/pull/1331, which uses this derived field.

### CODE REVIEW

File: esmvalcore/preprocessor/_multimodel.py [hunk 1]
```diff
@@ -107,15 +107,22 @@ def _put_in_cube(template_cube, cube_data, statistic, t_axis):
             t_axis,
             standard_name='time',
             units=template_cube.coord('time').units)
-    lats = template_cube.coord('latitude')
-    lons = template_cube.coord('longitude')
+
```
Comment (bouweandela) [inline esmvalcore/preprocessor/_multimodel.py]: These changes seem unrelated to deriving the variable? Would it make sense to put this in a separate pull request?

File: esmvalcore/preprocessor/_derive/intppgt.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    @staticmethod
+    def calculate(cubes):
+        """Compute longwave cloud radiative effect."""
+        intpp_cube = cubes.extract_strict(
+            iris.Constraint(name='net_primary_mole_productivity_of_carbon_by_'
+                                 'phytoplankton'))
+
+        try:
+            cube_area = cubes.extract_strict(iris.Constraint(name='cell_area'))
+        except iris.exceptions.ConstraintMismatchError:
```
Comment (bouweandela) [inline esmvalcore/preprocessor/_derive/intppgt.py]: If this happens, `cube_area` will be undefined and the code below will fail, so I think it is not a good idea to let this `pass`

File: esmvalcore/preprocessor/_derive/intppgt.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+        total_flux = calculate_total_flux(intpp_cube, cube_area)
+
+        # Dummy result cube
+        result = intpp_cube.collapsed(
+            ['latitude', 'longitude'],
+            iris.analysis.MEAN,
+        )
+
+        # TODO: Load seconds per year from model calendar.
```
Comment (bouweandela) [inline esmvalcore/preprocessor/_derive/intppgt.py]: Do your to do?

File: esmvalcore/preprocessor/_derive/intppgt.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    Returns
+    -------
+    numpy.array:
+        An numpy array containing the total flux of CO2.
+    """
+    data = []
+    times = intpp_cube.coord('time')
+
+    intpp_cube.data = np.ma.array(intpp_cube.data)
+    for time_itr in np.arange(len(times.points)):
```
Comment (bouweandela) [inline esmvalcore/preprocessor/_derive/intppgt.py]: Would it be possible to use `iris.analysis.SUM` to aggregate over the required dimensions?

**Comment by mattiarighi** [comment]
@ledm can you finalize this PR so that I can test it?

**Review by znicholls — COMMENTED** [review]
Following the logic of https://github.com/ESMValGroup/ESMValCore/pull/331, is this now redundant? This could be specified in a recipe using a combination of fx weighting, the area information and the sum preprocessor?

This PR can thus be closed?

**Comment by znicholls** [comment]
The fact that the derived variable's units lose the per m2 is related to the comment here https://github.com/ESMValGroup/ESMValCore/pull/1574#issuecomment-1123428679

In other words, the fact that a sum weighted by area does not include units (i.e. by multiplying the units by area ** 2) is an ongoing issue (issue now open at #1613 to track)

**Comment by schlunma** [comment]
I think this can now be done with the following preprocessor:

```yaml
area_statistics:
  operator: sum
```

(including correct unit handling).

Please re-open if necessary.
